src/metabase/query_processor/middleware/annotate.clj

```diff
@@ -193,7 +193,8 @@
         ;; remapping based on that info)
         namespaced-options        (not-empty (into {}
                                                    (filter (fn [[k _v]]
-                                                             (and (keyword? k) (namespace k))))
+                                                             (and (keyword? k) (namespace k)
+                                                                  (not= k :metabase.query-processor.middleware.mark-outer-select-fields/outer-select))))
```

Comment:
I'm not quite sure what the logic is behind `namespaced-options`. I'm removing this "annotation" because it's really only needed during compilation and leaving it here makes a lot of tests fail.

Reply:
I think we should check if we can get the tests passing. Perhaps we can do some judicious use of `partial=` to handle new keys? As it stands, this code makes me think its "special" rather than just unnecessary and removed for tests. Do you remember how many tests were failing?

Reply:
I would expect them to be different in exactly the current circumstance. Suppose `foo` is an integer column that has a `:Coercion/UNIXSeconds->DateTime` coercion on it. I would expect the query to be

`select foo from (select from_unixtime(foo) from t)`

I think we have to do the coercion at its lowest level. We really want no part of the query to see the integer of `foo`, it should be a timestamp type value throughout.

Reply:
I think you misunderstood what I meant. In your example what I wanted to say was that I expected the metadata to be the same for `select from_unixtime(foo) as foo from t` and `select foo from (select from_unixtime(foo) as foo from t)`. In other words, just wrapping a select `s` into a `select * from (s)` should not "change" the metadata.